ml-agents/mlagents/envs/environment.py

```diff
@@ -361,17 +353,19 @@ def step(self,  vector_action=None, memory=None, text_action=None, value=None) -
                         "The brain {0} expected {1} text_action but was given {2}".format(
                             b, n_agent, len(text_action[b])))
                 if not ((self._brains[b].vector_action_space_type == "discrete" and len(
-                        vector_action[b]) == n_agent * len(self._brains[b].vector_action_space_size)) or
-                            (self._brains[b].vector_action_space_type == "continuous" and len(
-                                vector_action[b]) == self._brains[b].vector_action_space_size[0] * n_agent)):
+                        vector_action[b]) == n_agent * len(
```

Comment:
this block condition is super confusing ... can we use intermediate variables to separate the different checks.

For example:
```
total_space_size = n_agent * len(self._brains[b].vector_action_space_size)
discrete_space_check = (self._brains[b].vector_action_space_type == "discrete" and len(vector_action[b]) == total_space_size
if not (discrete_space_check or ...
```

Reply:
These temp vars can now be reused in the error messages that follow.

Reply:
Re-write / re-organize block till line 368 for readability.